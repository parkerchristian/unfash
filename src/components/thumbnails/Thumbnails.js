import React from 'react';
import PropTypes from 'prop-types';
import Thumbnail from './Thumbnail';

function Thumbnails({ thumbnails }) {
  const thumbnailList = thumbnails.map(thumbnail => (
    <li key={thumbnail.imgUrl}>
      <Thumbnail imgUrl={thumbnail.imgUrl} title={thumbnail.title} />
    </li>
  ));
  return (
    <ul>{thumbnailList}</ul>
  );
}

Thumbnails.propTypes = {
  thumbnails: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Thumbnails;
