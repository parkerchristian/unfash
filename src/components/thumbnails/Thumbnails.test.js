import React from 'react';
import Thumbnails from './Thumbnails';
import { shallow } from 'enzyme';

describe('Thumbnails component', () => {
  const thumbnailList = [
    {
      imgUrl: 'https://images.unsplash.com/photo-1556229174-5e42a09e45af?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=643&q=80',
      title: 'One dog'
    },
    {
      imgUrl: 'https://images.unsplash.com/photo-1559032534-1a20db37cdd2?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=634&q=80',
      title: 'Two Dogs.'
    },
    {
      imgUrl: 'https://images.unsplash.com/photo-1558945657-484aa38065ec?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=633&q=80',
      title: 'and a third'
    }
  ];
  it('Renders a list of thumbnails', ()=> {
    const wrapper = shallow(<Thumbnails thumbnails={thumbnailList} />);
    expect(wrapper).toMatchSnapshot();
  });
});
