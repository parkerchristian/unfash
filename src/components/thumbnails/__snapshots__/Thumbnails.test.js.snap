// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails component Renders a list of thumbnails 1`] = `
<ul>
  <li>
    <Thumbnail
      imgUrl="https://images.unsplash.com/photo-1556229174-5e42a09e45af?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=643&q=80"
      title="One dog"
    />
  </li>
  <li>
    <Thumbnail
      imgUrl="https://images.unsplash.com/photo-1559032534-1a20db37cdd2?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=634&q=80"
      title="Two Dogs."
    />
  </li>
  <li>
    <Thumbnail
      imgUrl="https://images.unsplash.com/photo-1558945657-484aa38065ec?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=633&q=80"
      title="and a third"
    />
  </li>
</ul>
`;
